web-app/src/main/webapp/script.js

```diff
@@ -156,19 +156,20 @@ function hideReportForm() {
 
 // This currently gets the address from the report form's location field (no autopopulate, no map picker)
 async function postUserReport(geocoder) {
-  document.getElementById('report-form').reset();
-
   const address = document.getElementById('location-input').value;
   geocoder.geocode({ 'address': address }, async (results, status) => {
     if (status === 'OK') {
       const coordinates = results[0].geometry.location;
       const data = reportFormToURLQuery(coordinates.lat(), coordinates.lng());
       const url = await fetchBlobstoreUrl();
       fetch(url, { method: 'POST', body: data });
+      document.getElementById('report-form').reset();
     } else {
       console.error('Geocode was not successful: ' + status);
     }
   })
+
+  hideReportForm();
```

Comment:
This is still done immediately after the geocode request is sent, right?

Reply:
Modified so that we only hide the report form once geocoding is done successfully and we confirm that all fields are filled in.